MonkeyLoader.GamePacks.ResoniteModLoader/ModConfigurationDefinitionBuilder.cs

```diff
@@ -95,6 +104,16 @@ private void ProcessField(FieldInfo field)
             }
 
             var fieldValue = (ModConfigurationKey)field.GetValue(field.IsStatic ? null : _owner);
+
+            if (field.GetCustomAttribute<RangeAttribute>() is RangeAttribute rangeAttribute)
+            {
+                _owner.Logger.Info(() => $"Found FrooxEngine.RangeAttribute on field {field.Name}");
+                var min = Convert.ChangeType(rangeAttribute.Min, fieldValue.ValueType());
+                var max = Convert.ChangeType(rangeAttribute.Max, fieldValue.ValueType());
```

Comment:
You could do this conversion in the AddRangeComponent method, and just pass in the RangeAttribute instance instead of the concrete min/max. Might make the type changing easier?

Reply:
I don't think that helps, the casting still doesn't work:

```
private static void AddRangeComponent<T>(DefiningConfigKey<T> key, RangeAttribute rangeAttribute)
{
    var rangeComponent = new ConfigKeyRange<T>((T)(object)rangeAttribute.Min, (T)(object)rangeAttribute.Max, null);
    key.Components.Add(rangeComponent);
}
```